### ISSUE TITLE
how to change previewheight automatically

### ISSUE THREAD

**Issue by rainerborene** [Shougo/unite.vim#824]
I want to change `previewheight` when opening Unite and revert to default value on close. It's possible to do that?

**Comment by rainerborene** [comment]
Ok. This is working for me:

```viml
function! s:unite_settings()
  setlocal previewheight=50  
endfunction
autocmd FileType unite call s:unite_settings()
autocmd FileType unite autocmd BufHidden <buffer> setlocal previewheight&
```

I didn't realize you could nest autocommands like that.

**Comment by rainerborene** [comment]
Actually `unite_settings` is called only once when using the `resume` option.

**Comment by Shougo** [comment]
OK.  I will put it in todo list.

**Comment by rainerborene** [comment]
If you give me the directions I could help you with that and send a pull request.

**Comment by Shougo** [comment]
You should change `unite#view#_preview_file()` function.

**Comment by rainerborene** [comment]
Have a look on this video. https://www.dropbox.com/s/0282yi05brobalj/unite_bug.ogv?dl=0

Download the video your local machine because the preview quality is crap. As you can see the Unite window decreases the height automatically. Any idea why this is happening? Btw, I didn't make any modification to `unite.vim` yet. Here's my [.vimrc](https://github.com/rainerborene/dotfiles/blob/master/.vimrc)

Steps to reproduce the bug:

- `:Unite -no-split -buffer-name=grep -no-empty -resume -auto-preview grep`
- Search for anything.
- Close unite with `q` keybinding.
- Execute the first step again.
- Do another search using `<C-r>`
- Navigate using the `j` and `k` letters.

**Comment by Shougo** [comment]
I get it.  It is bug.
@rainerborene Please create new issue for it.  It is not bug report issue.

**Comment by Shougo** [comment]
Implemented.
